### PR TITLE
Dockerfile improvement

### PR DESCRIPTION
**Pull request by d-huck**
This PR adds some quality of life improvements to the Dockerfile recently added to the repo.

First, the `-j` flag on the make command for `ut_automata` would fail when building on macOS. While this may be limited to my machine, the time savings is minimal from adding the flag and was removed.

Second, this installs `tmux` in the container and introduces a simple launcher script that will create a tmux session with the entire infrastructure stack running. Students should only have to run `make docker_shell` to enter into the docker container. From there, `roscore`, `simulator`, and `websocket` are already running in a detached tmux session.

### CODE REVIEW

**Comment by d-huck** [comment]
Generalized the tmux script so it can also be used on UT machines with an `$INSTALL_PATH$ variable. If not provided, it will error out and stop. Added context variables to see if tmux script is running inside the docker container or on a bare machine.

**Comment by d-huck** [comment]
Finished changes, should be ready for review.

File: Dockerfile [hunk 1]
```diff
@@ -37,4 +40,11 @@ RUN echo -e "source /opt/ros/noetic/setup.bash\n" \
 
 # build deps
 RUN source ~/.profile && cd amrl_msgs && make -j
-RUN source ~/.profile && cd ut_automata && make -j
+RUN source ~/.profile && cd ut_automata && make
```
Comment (rmeno12) [inline Dockerfile]: can you elaborate on what the error you got was? I'm curious why it affected ut_automata and not amrl_msgs

File: Dockerfile [hunk 2]
```diff
@@ -5,7 +5,10 @@ RUN apt-get update && \
     apt-get install -y git libgflags-dev libpopt-dev \
                        libgoogle-glog-dev liblua5.1-0-dev \
                        libboost-all-dev libqt5websockets5-dev \
-                       python-is-python3 libeigen3-dev sudo
+                       python-is-python3 libeigen3-dev sudo 
+
+# multiplexers to run roscore and other infra in background
+RUN apt-get install -y tmux
```
Comment (rmeno12) [inline Dockerfile]: combine this with the previous apt command to minimize layers

**Review by rmeno12 — CHANGES_REQUESTED** [review]
nice job with the tmux setup, works smoothly

File: Dockerfile (hunk 1 above)
Comment (d-huck) [reply Dockerfile]: I was getting linker errors on the build. I believe it was trying to link an object that wasn't finished compiling yet. It only occurs on my Macbook, but runs fine on my Linux box. It was throwing the same error when I was building a separate Dockerfile before I saw the original Dockerfile PR. May be worthwhile to take the `-j` off of `amrl_msg` as well for safety.

File: Dockerfile (hunk 2 above)
Comment (d-huck) [reply Dockerfile]: Done.

**Review by joydeep-b — APPROVED** [review]
Looks good to me, thanks! @rmeno12 please merge when ready.

File: Dockerfile (hunk 1 above)
Comment (rmeno12) [reply Dockerfile]: alright go ahead and remove it from there as well, then we can merge it in

**Review by rmeno12 — APPROVED** [review]

**Comment by joydeep-b** [comment]
Anything missing on this? Is it ready to merge?

**Comment by rmeno12** [comment]
I believe there's a change to the dockerfile needed to not build amrl_msgs with `-j`. Also @d-huck mentioned to me in person that there were issues with the CI as a result of the tmux entrypoint.

**Comment by d-huck** [comment]
Hi @rmeno12 @joydeep-b,

Apologies this got away from me, phase 2 of the course is not as kind to us CS folks as phase 1 was.

I fastforwarded the changes in the docker file from #7 and this now passes all build tests. I also removed the `-j` from both make commands for `ut_automata` and `amrl_msgs` since these cause failure on (at least my) M1 Mac, and the speed up from `-j` is amortized once the Dockerfile has been built once.

**Review by rmeno12 — CHANGES_REQUESTED** [review]
Can you update the readme so the new workflow is documented on there?

**Comment by rmeno12** [comment]
Looks good! can you either merge master into your branch or rebase it so that the conflicts are resolved?

**Review by rmeno12 — APPROVED** [review]

**Pull request merged**